src/crypto/public_key/dsa.js

```diff
@@ -148,25 +148,24 @@ export default {
     /**
      * Check that subgroup order q divides p-1
      */
-    if (!p.sub(one).mod(q).isZero()) {
+    if (!p.dec().mod(q).isZero()) {
       return false;
     }
 
-    const pred = new BN.red(p);
-    const gModP = g.toRed(pred);
     /**
      * g has order q
      * Check that g ** q = 1 mod p
      */
-    if (!gModP.redPow(q).eq(one)) {
+    if (!g.modExp(q, p).isOne()) {
       return false;
     }
 
     /**
      * Check q is large and probably prime (we mainly want to avoid small factors)
      */
-    const qSize = q.bitLength();
-    if (qSize < 150 || !(await prime.isProbablePrime(q, null, 32))) {
+    const qSize = new BigInteger(q.bitLength());
+    const n150 = new BigInteger(150);
+    if (qSize.lt(n150) || !(await prime.isProbablePrime(q, null, 32))) {
```

Comment:
These don't need to be `BigInteger`s, right? Since both are coming from a `Number`, they're guaranteed to fit in a `Number`